### PR TITLE
Don't send a status update when successful

### PR DESCRIPTION
**Pull request by kenoir**

### CODE REVIEW

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifierWorkerService.scala [hunk 1]
```diff
@@ -3,31 +3,21 @@ package uk.ac.wellcome.platform.archive.bagverifier.services
...
 class BagVerifierWorkerService(
-  notificationStream: NotificationStream[BagRequest],
-  verifyDigestFilesService: VerifyDigestFilesService,
-  progressSnsWriter: SNSWriter,
-  outgoingSnsWriter: SNSWriter,
-)(implicit ec: ExecutionContext)
-    extends Runnable
+                                notificationStream: NotificationStream[BagRequest],
+                                verifyDigestFilesService: VerifyDigestFilesService,
+                                notificationService: NotificationService
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifierWorkerService.scala]: Naming nit: having a “notificationStream” that’s an input and a “notificationService” that’s an output feels liable to get confusing. Could we rename them to emphasise the input/output nature?

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import uk.ac.wellcome.platform.archive.common.models.BagRequest
+import uk.ac.wellcome.platform.archive.common.progress.models.{Progress, ProgressNotice, ProgressUpdate}
+import uk.ac.wellcome.json.JsonUtil._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class NotificationService(
+                           progressSnsWriter: SNSWriter,
+                           outgoingSnsWriter: SNSWriter
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: Naming nit: both of these writers are outgoing; I think we use “ongoing” in some other places?

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        info(
+          summarizeVerification(
+            bagRequest,
+            bagVerification
+          )
+        )
+
+        if (bagVerification.verificationSucceeded) {
+          ProgressNotice(
+            bagRequest.archiveRequestId,
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: Keyword arguments please!

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  private def summarizeVerification(
+                                     bagRequest: BagRequest,
+                                     bagVerification: BagVerification): String = {
+    val verificationStatus = if (bagVerification.verificationSucceeded) {
+      "successful"
+    } else {
+      "failed"
+    }
+    f"""$verificationStatus verification
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: Don’t you want an s-string here?

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala (hunk 2 above)
Comment (kenoir) [reply bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: Arguably both are ongoing too - we reference ongoing in terraform only, so i'll update that to outgoing so it's consistent.

Comment (alexwlchan) [reply bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: You've also used ongoing elsewhere in this patch!

I wonder if “pipeline” might be a better word?

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifierWorkerService.scala [hunk 5]
```diff
@@ -36,97 +26,38 @@ class BagVerifierWorkerService(
...
+      )
+
+      _ <- notificationService
+        .sendProgressNotification(
+          bagRequest,
+          tryBagVerification
+        )
+
+      _ <- notificationService
+        .sendOngoingNotification(
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifierWorkerService.scala]: 👉

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala (hunk 3 above)
Comment (kenoir) [reply bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/NotificationService.scala]: 👍

**Pull request merged**